### PR TITLE
chore: refactor checkDSL and add in validation infrastucture

### PR DESCRIPTION
**Pull request by adriantam**
## Description
Refactor checkDSL and add in validation test infrastucture in preparation for additional validation for list object.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: src/check-dsl.ts [hunk 1]
```diff
@@ -27,109 +41,144 @@ const defaultError = (lines: any) => {
...
-        const { relation: relationName } = relationDef;
+      encounteredRelationsInType[relationName] = true;
+      globalRelations[relationName] = true;
+    });
+    transformedTypes[typeName] = {
+      comment: "",
+      type: typeName,
+      relations,
+    };
+  });
```
Comment (miparnisari) [inline src/check-dsl.ts:90]:
```suggestion
  return [globalRelations, transformedTypes] as const
  });
  ```
  and get rid of the function parameters?

Comment (adriantam) [reply src/check-dsl.ts:90]: done

File: src/check-dsl.ts [hunk 2]
```diff
@@ -27,109 +41,144 @@ const defaultError = (lines: any) => {
...
-        if (relationDef.definition.base) {
-          validateTargetRelation(typeName, relationName, relationDef.definition.base);
-        }
-        if (relationDef.definition.diff) {
-          validateTargetRelation(typeName, relationName, relationDef.definition.diff);
-        }
-      });
-    });
+    populateRelation(lines, reporter, parserResults, globalRelations, transformedTypes);
+    basicValidateRelation(lines, reporter, parserResults, transformedTypes, globalRelations);
```
[the remark this answers is not in the document]
Comment (adriantam) [reply src/check-dsl.ts]: done

File: src/check-dsl.ts [hunk 3]
```diff
@@ -27,109 +41,144 @@ const defaultError = (lines: any) => {
   };
 };
 
-export const checkDSL = (codeInEditor: string) => {
-  const lines = codeInEditor.split("\n");
-  const markers: any = [];
-  const reporter = report({ lines, markers });
+// helper function to populate the relationsPerType and globalRelations list
+// will throw error if there are duplication relations
+const populateRelation = (
```
[the remark this answers is not in the document]
Comment (adriantam) [reply src/check-dsl.ts]: done

File: src/check-dsl.ts [hunk 4]
```diff
@@ -27,109 +41,144 @@ const defaultError = (lines: any) => {
   };
 };
 
-export const checkDSL = (codeInEditor: string) => {
-  const lines = codeInEditor.split("\n");
-  const markers: any = [];
-  const reporter = report({ lines, markers });
+// helper function to populate the relationsPerType and globalRelations list
+// will throw error if there are duplication relations
```
[the remark this answers is not in the document]
Comment (adriantam) [reply src/check-dsl.ts]: updated

**Review by miparnisari — APPROVED** [review]

**Pull request merged**